src/main/java/software/amazon/awssdk/crt/s3/S3Client.java

```diff
@@ -65,12 +67,19 @@ public S3Client(S3ClientOptions options) throws CrtRuntimeException {
             monitoringThroughputThresholdInBytesPerSecond = monitoringOptions.getMinThroughputBytesPerSecond();
             monitoringFailureIntervalInSeconds = monitoringOptions.getAllowableThroughputFailureIntervalSeconds();
         }
+        AwsSigningConfig signingConfig = options.getSigningConfig();
+        boolean created_signing_config = false;
+        if(signingConfig == null) {
```

Comment:
Trivial: The name `created_signing_config` seems a bit strange to me (you can keep it if you like it). Maybe we could update it to something like `did_create_signing_config` or `is_signing_config_present`, or we could even eliminate this variable entirely and directly use `options.getSigningConfig() == null` check later.